drivers/sensor/si7006/si7006.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2019 Vaishali Pathak <vaishali@electronut.in>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sensor.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <string.h>
+#include <misc/byteorder.h>
+#include <misc/__assert.h>
+#include <logging/log.h>
+#include <i2c.h>
+#include <logging/log.h>
+#include <stdio.h>
+#include "si7006.h"
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_REGISTER(si7006);
+
+struct si7006_data {
+	struct device *i2c_dev;
+	s32_t temperature_val1;
+	s32_t temperature_val2;
+	s32_t humidity_val1;
+	s32_t humidity_val2;
+};
+
+/**
+ * @brief function to get relative humidity
+ *
+ * @return int 0 on success
+ */
+static int si7006_get_humidity(struct device *i2c_dev,
+	struct si7006_data *si_data)
+{
+	int retval = 0;
+	uint8_t hum[2];
+
+	if (i2c_burst_read(i2c_dev, DT_SILABS_SI7006_0_BASE_ADDRESS,
+			SI7006_MEAS_REL_HUMIDITY_MASTER_MODE, hum, sizeof(hum))
+			!= 0) {
+		LOG_ERR("read register err");
+		retval = -1;
+	}
+
+	uint16_t relative_humidity_code = (hum[0] << 8) | (hum[1]);
+
+	si_data->humidity_val1 = ((125 * relative_humidity_code) >> 16) - 6;
+	si_data->humidity_val2 = 0;
```

Comment:
You're throwing away a lot of precision with the 16-bit right shift. Are you sure `humidity_val2` should really be zero?

I also suggest storing the raw sensor data (`relative_humidity_code` and `temperature_code`) in `si7006_data`, then do the SI conversion in the `channel_get` function. This will save some ram and better follow the sensor API.

Reply:
Couldn't work on this earlier, I was busy.

> You're throwing away a lot of precision with the 16-bit right shift. Are you sure `humidity_val2` should really be zero?
> The relative humidity value is in percentage and is always whole number.
![si7--6_humidity](https://user-images.githubusercontent.com/33493760/60255190-e39e0a80-98ec-11e9-960d-94201d8bdea2.png)
But, I made changes, I'm not assigning  0 value to the val2 of humidity. Now, it will be calculated.

> I also suggest storing the raw sensor data (`relative_humidity_code` and `temperature_code`) in `si7006_data`, then do the SI conversion in the `channel_get` function. This will save some ram and better follow the sensor API.
Done this!

Reply:
Ok. Could you please add a comment referencing the datasheet so we don't have the same questions in the future?